flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/compiler/NFACompiler.java

```diff
@@ -74,6 +77,36 @@
 		}
 	}
 
+	/**
+	 * Verifies if the provided pattern can possibly generate empty match. Example of patterns that can possibly
+	 * generate empty matches are: A*, A?, A* B? etc.
+	 *
+	 * @param pattern pattern to check
+	 * @return true if empty match could potentially match the pattern, false otherwise
+	 */
+	public static boolean canProduceEmptyMatches(final Pattern<?, ?> pattern) {
+		NFAFactoryCompiler<?> compiler = new NFAFactoryCompiler<>(checkNotNull(pattern));
+		compiler.compileFactory();
+		State<?> startState = compiler.getStates().stream().filter(State::isStart).findFirst().get();
```

Comment:
nit: `orElseThrow(IllegalStateException::new)` (or sth like this, I'm not sure what's the correct syntax) instead of `get()`?

Reply:
Ok, will change it. In general though I am always torn in this situations, cause that would mean we have a bug in our compiler.

Reply:
But that's the convention: you `checkState`, `checkArgument`, throw `IllegalStateException` or `IllegalArgumentException` to cover for bugs in the code - the situations/class states that should never happen.